exec/cnex/cnex.c

```diff
@@ -154,45 +196,113 @@ int main(int argc, char* argv[])
                         exec->diagnostics.total_opcodes,
                         exec->stack->max + 1,
                         exec->stack->top,
-                        exec->diagnostics.callstack_max_height + 1, 
-                        exec->callstacktop, 
-                        exec->object->global_size, 
+                        exec->diagnostics.callstack_max_height + 1,
+                        exec->callstacktop,
+                        exec->module[0].bytecode->global_size,
                         exec->framestack->max,
                         (((float)exec->diagnostics.time_end - exec->diagnostics.time_start) / CLOCKS_PER_SEC) * 1000
         );
     }
 
-    global_shutdown();
     exec_freeExecutor(exec);
-    bytecode_freeBytecode(pModule);
+    global_shutdown();
 
-    free(bytecode);
+    free(gOptions.pszExecutablePath);
 
     return 0;
 }
 
 int exec_run(TExecutor *self, BOOL enable_assert)
 {
-    self->ip = (uint32_t)self->object->codelen;
-    invoke(self, 0);
     self->enable_assert = enable_assert;
+    self->ip = (uint32_t)self->module->bytecode->codelen;
+    invoke(self, self->modules[0], 0);
+
+    for (unsigned int m = 1; m < self->module_count; m++) {
```

Comment:
I'm not sure this executes the module initialization code in the correct order. It's important to get this right. How is your code different from `exec.cpp` here?

Reply:
It executes the code in the order in which the modules are discovered in code.  I thought that was how `exec.cpp` did it as well?  Perhaps I need to go back and study `exec.cpp` more.

Reply:
Fixed with the addition of the test case for testing the load order.

Reply:
Fixed.  Modules are now initialized in the correct order, the order they were loaded/referenced in.